fvm/fvm.go

```diff
@@ -16,17 +16,36 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
+type ProcedureType string
+
+const (
+	BootstrapProcedureType   = ProcedureType("bootstrap")
+	TransactionProcedureType = ProcedureType("transaction")
+	ScriptProcedureType      = ProcedureType("script")
+)
+
 // An Procedure is an operation (or set of operations) that reads or writes ledger state.
 type Procedure interface {
 	Run(
 		ctx Context,
 		sth *state.StateHolder,
-		programs *programs.Programs,
+		programs *programs.TransactionPrograms,
 	) error
 
 	ComputationLimit(ctx Context) uint64
 	MemoryLimit(ctx Context) uint64
 	ShouldDisableMemoryAndInteractionLimits(ctx Context) bool
+
+	Type() ProcedureType
+
+	// The initial snapshot time.  Note that once we start supporting parallel
+	// preprocessing/execution, a transaction may operation on mutliple
+	// snapshots.  For scripts, the snapshot time is EndOfBlockExecutionTime.
+	InitialSnapshotTime() programs.LogicalTime
```

Comment:
```suggestion
	// The initial snapshot time.  Note that once we start supporting parallel
	// preprocessing/execution, a transaction may operation on multiple
	// snapshots.  
	// For scripts, since they can only be executed after the block has been executed, 
	// the initial snapshot time for scripts is EndOfBlockExecutionTime.
	InitialSnapshotTime() programs.LogicalTime
```
Could you explain what would the following behavior to do with the initial snapshot time?
> Note that once we start supporting parallel
	// preprocessing/execution, a transaction may operation on multiple
	// snapshots.

Reply:
added more comment